### ISSUE TITLE
LSP giving unrelated completion suggestions in situations like after a keyword, after \,  in comments, or on a variable of basic types

### ISSUE THREAD

**Issue by JoannesJ** [godotengine/godot#69912]
### Godot version

v4.0.beta7.official [0bb1e89fb]

### System information

Ubuntu 22.10

### Issue description

In certain situations the LSP will throw over a giant load of unrelated completion suggestions. It seems like it is suggesting everything.
It seems like this load of suggestions is responsible for some slow down when using the LSP.
I've tested this with nvim's lsp and vscode.
Some situations where this happens is after a keyword, in a comment, and after a `\`.
It also happens when completing on properties of a basic(int, bool) or `any` type.
In contrast Godot's text editor doesn't provide any suggestions in these situations if completion is triggered on no text, otherwise it gives suggestions related to the script.

In short it seems that the LSP is throwing over everything where completions don't make sens.

### Steps to reproduce

In any script on an external editor with LSP, trigger code completion in any of the situations described above.
Notice how there are suggestions being made to properties of unrelated classes.

### Minimal reproduction project

N/A
